activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java

```diff
@@ -89,6 +91,10 @@ public TaskRuntimeConfiguration configuration() {
     public Task task(String taskId) {
         return taskConverter.fromWithCandidates(taskRuntimeHelper.getInternalTaskWithChecks(taskId));
     }
+    
+    private Task reassignedTask(String taskId) {
+        return taskConverter.fromWithCandidates(taskRuntimeHelper.getInternalTask(taskId));
+    }
```

Comment:
When the user reassign a task, he is not anymore the assignee, for this reason the method `getInternalTaskWithChecks` fails to find the task. I used `getInternalTask` as workaround but another solution (?) could be change the implementation of [getInternalTaskWithChecks](https://github.com/Activiti/Activiti/blob/eeda8439bcd30f525f10b23f54a7cbb4819d2ef9/activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeHelper.java#L190) with less restrictive chcks.

Reply:
As discussed, the API `getInternalTaskWithChecks` works for candidate users only if the current assignee is null [(see query)](https://github.com/Activiti/Activiti/blob/2d94c7d0f3be62a1abef9da8a53f9e7dfa0f0b2c/activiti-core/activiti-engine/src/main/resources/org/activiti/db/mapping/entity/Task.xml#L602). In my case after the assign operation, the assignee is not null and the current user is still a candidate but the API doesn't return the task. Because of that I will use `getInternalTask`.